src/widget/wwaveformviewer.cpp

```diff
@@ -184,11 +184,25 @@ void WWaveformViewer::slotTrackLoaded(TrackPointer track) {
 }
 
 void WWaveformViewer::slotLoadingTrack(TrackPointer pNewTrack, TrackPointer pOldTrack) {
-    Q_UNUSED(pNewTrack);
-    Q_UNUSED(pOldTrack);
+    if (pOldTrack) {
+        disconnect(pOldTrack.get(), SIGNAL(waveformUpdated()),
+                   this, SLOT(slotWaveformUpdated()));
+    }
+
     if (m_waveformWidget) {
         m_waveformWidget->setTrack(TrackPointer());
     }
+
+    if (pNewTrack) {
+        connect(pNewTrack.get(), SIGNAL(waveformUpdated()),
+                this, SLOT(slotWaveformUpdated()));
```

Comment:
the track was just cleared above. Can we move this to slotTrackLoaded, to keep symmetry?